integration/test/createnodepool/createnodepool_test.go

```diff
@@ -0,0 +1,142 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	expcapiv1alpha3 "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/integration/env"
+	"github.com/giantswarm/azure-admission-controller/integration/util"
+	"github.com/giantswarm/azure-admission-controller/integration/values"
+)
+
+const (
+	crsFolder       = "testdata"
+	prodCatalogName = "control-plane-catalog"
+	testCatalogName = "control-plane-test-catalog"
+	// API Groups for upstream Cluster API types.
+	giantswarmCoreAPIGroup             = "core.giantswarm.io"
+	clusterAPIGroup                    = "cluster.x-k8s.io"
+	infrastructureAPIGroup             = "infrastructure.cluster.x-k8s.io"
+	experimentalClusterAPIGroup        = "exp.cluster.x-k8s.io"
+	experimentalInfrastructureAPIGroup = "exp.infrastructure.cluster.x-k8s.io"
+	securityAPIGroup                   = "security.giantswarm.io"
+)
+
+func TestCreateCluster(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var appTest apptest.Interface
+	{
+		runtimeScheme := runtime.NewScheme()
+		appSchemeBuilder := runtime.SchemeBuilder{
+			applicationv1alpha1.AddToScheme,
+			apiextensionsv1.AddToScheme,
+			capiv1alpha3.AddToScheme,
+			capzv1alpha3.AddToScheme,
+			expcapiv1alpha3.AddToScheme,
+			expcapzv1alpha3.AddToScheme,
+			securityv1alpha1.AddToScheme,
+			corev1.AddToScheme,
+			corev1alpha1.AddToScheme,
+		}
+		err = appSchemeBuilder.AddToScheme(runtimeScheme)
+		if err != nil {
+			t.Fatal(err)
+		}
+		appTest, err = apptest.New(apptest.Config{
+			KubeConfigPath: env.KubeConfig(),
+			Logger:         logger,
+			Scheme:         runtimeScheme,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	{
+		err = appTest.EnsureCRDs(ctx, getRequiredCRDs())
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	{
+		valuesYAML, err := values.YAML(env.AzureClientID(), env.AzureClientSecret(), env.AzureSubscriptionID(), env.AzureTenantID())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		apps := []apptest.App{
+			{
+				CatalogName:   prodCatalogName,
+				Name:          "cert-manager-app",
+				Namespace:     metav1.NamespaceDefault,
+				Version:       "2.3.1",
+				WaitForDeploy: true,
+			},
+			{
+				CatalogName:   testCatalogName,
+				Name:          "azure-admission-controller",
+				Namespace:     metav1.NamespaceDefault,
+				SHA:           env.CircleSHA(),
+				ValuesYAML:    valuesYAML,
+				WaitForDeploy: true,
+			},
+		}
+		err = appTest.InstallApps(ctx, apps)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Give time to the admission controller to start.
+	time.Sleep(time.Second * 60)
```

Comment:
can't we do it a little bit smarter? there are a ton of reasons why this could fail.